packages/fastify-renderer/src/node/renderers/react/ReactRenderer.tsx

```diff
@@ -187,19 +187,22 @@ export class ReactRenderer implements Renderer {
     // push the props for the entrypoint to use when hydrating client side
     bus.push('tail', `<script type="module">window.__FSTR_PROPS=${JSON.stringify(render.props)};</script>`)
 
-    const entrypointName = this.buildVirtualClientEntrypointModuleID(render)
-    // if we're in development, we just source the entrypoint directly from vite and let the browser do its thing importing all the referenced stuff
-    if (this.plugin.devMode) {
-      bus.push(
-        'tail',
-        `<script type="module" src="${path.join(
-          this.plugin.assetsHost,
-          this.entrypointScriptTagSrcForClient(render)
-        )}"></script>`
-      )
-    } else {
-      const manifestEntryName = normalizePath(path.relative(this.viteConfig.root, entrypointName))
-      this.plugin.pushImportTagsFromManifest(bus, manifestEntryName)
+    // If the route is not imperative, we want to hydrate it on client-side
+    if (!render.isImperative) {
```

Comment:
I think hydration is still really important so that any react code which has user interactions works on the page, no? Without this, components like a navigation or an auth account selector or what have you just won't work for imperatively rendered pages. I think that's a no go. Did you need ti disable hydration because the client side router didn't have the route for the imperative page? My guess is yes. Perhaps for imperatively rendered components, we could send an entrypoint that has the extra route?